[DOC] Update the Pyroscope data source config doc (#89648)
grafana/grafana: docs/sources/administration/provisioning/index.md

#### Custom HTTP headers for data sources

> Data sources managed by Grafana's provisioning can be configured to add HTTP headers to all requests
> going to that data source. The header name is configured in the `jsonData` field and the header value should be
> configured in `secureJsonData`.

Reviewer: https://github.com/grafana/writers-toolkit/blob/main/vale/Grafana/ProductPossessives.yml#L4-L6

```suggestion
Data sources managed with provisioning can be configured to add HTTP headers to all requests.
The header name is configured in the `jsonData` field and the header value should be configured in `secureJsonData`.
```

#### Custom HTTP headers for data sources

Data sources managed with provisioning can be configured to add HTTP headers to all requests.

> The header name is configured in the `jsonData` field and the header value should be configured in `secureJsonData`.

Reviewer 2: Should be configured? or is configured Should always feels optional...

Author: ```suggestion
The header name is configured in the `jsonData` field and the header value is configured in `secureJsonData`.
```
